machine1/dbus.go

```diff
@@ -75,6 +77,75 @@ func (c *Conn) initConnection() error {
 	return nil
 }
 
+// GetImage may be used to get the image object path for the image with the specified name
+func (c *Conn) GetImage(name string) (dbus.ObjectPath, error) {
+	var path dbus.ObjectPath
+	result := c.object.Call(dbusInterface+".GetImage", 0, name)
+	if result.Err != nil {
+		log.Print(result.Err)
+		return path, result.Err
+	}
+	pathstr, ok := result.Body[0].(dbus.ObjectPath)
+	if ok {
+		path = pathstr
+	} else {
+		return path, fmt.Errorf("Error converting '%v' to dbus.ObjectPath.", result.Body[0])
+	}
+	return path, nil
+}
+
+// GetMachine retrieves the machine object path from dbus
+func (c *Conn) GetMachine(name string) (dbus.ObjectPath, error) {
+	var path dbus.ObjectPath
+	result := c.object.Call(dbusInterface+".GetMachine", 0, name)
+	if result.Err != nil {
+		log.Print(result.Err)
+		return path, result.Err
+	}
+	pathstr, ok := result.Body[0].(dbus.ObjectPath)
+	if ok {
+		path = pathstr
+	} else {
+		return path, fmt.Errorf("Error converting '%v' to dbus.ObjectPath.", result.Body[0])
+	}
+	return path, nil
+}
+
+// GetProperties retrieves the MachineProperties for any given machine
+func (c *Conn) GetProperties(name string) (map[string]interface{}, error) {
+	var machine map[string]interface{}
+	var path dbus.ObjectPath
+	var props map[string]dbus.Variant
+	result := c.object.Call(dbusInterface+".GetMachine", 0, name)
+	if result.Err != nil {
+		log.Print(result.Err)
+		return machine, result.Err
+	}
+	pathstr, ok := result.Body[0].(dbus.ObjectPath)
+	if ok {
+		path = pathstr
+	} else {
+		return machine, fmt.Errorf("Error converting '%v' to dbus.ObjectPath.", result.Body[0])
+	}
+	obj := c.conn.Object("org.freedesktop.machine1", path)
+	err := obj.Call("org.freedesktop.DBus.Properties.GetAll", 0, "").Store(&props)
+	if err != nil {
+		return machine, err
+	}
+
+	machine = make(map[string]interface{}, len(props))
+	for k, v := range props {
+		machine[k] = v.Value()
+	}
+	return machine, nil
+}
+
+// KillMachine signals systemd to power off the container
+func (c *Conn) KillMachine(name string) error {
+	var killsignal int32 = 38 // SIGTERMIN+4 (systemd power off)
```

Comment:
I stole it from machinectl.c

I expect future enhancements to allow setting the kill signal, but for my immediate needs this is sufficient.